frontend/src/main/utils/users.js

```diff
@@ -2,16 +2,60 @@ import { useQuery } from "react-query";
 import axios from "axios";
 
 export function useUsers() {
-  return useQuery("users", async () => {
-    const uri = "/api/admin/users";
-    try {
-      const response = await axios.get(uri);      
-      return response.data ;
-    } catch (e) {
-      console.error(`Error getting data from ${uri}:`,e);
-      return [];
+  return useQuery(
+    "users",
+    async () => {
+      const uri = "/api/admin/users";
+      try {
+        const response = await axios.get(uri);
+        return response.data;
+      } catch (e) {
+        console.error(`Error getting data from ${uri}:`, e);
+        return [];
+      }
+    },
+    {
+      initialData: [],
     }
-  }, {
-    initialData: []
-  });
+  );
+}
+
+export async function suspendUser(cell) {
+  const uri = "/api/admin/users/suspend";
+  const user = cell.row.values;
+  try {
+    const response = await axios.post(
+      uri,
+      {},
+      {
+        params: {
+          userId: user.id,
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    console.error(`Error suspending user with id ${user.id} from ${uri}`, e);
+    return [];
+  }
+}
+
+export async function restoreUser(cell) {
+  const uri = "/api/admin/users/restore";
+  const user = cell.row.values;
+  try {
+    const response = await axios.post(
+      uri,
+      {},
+      {
+        params: {
+          userId: user.id,
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    console.error(`Error restoring user with id ${user.id} from ${uri}`, e);
+    return [];
+  }
```

Comment:
Same here.  This should be a useBackendMutation or at least something that invalidates the react query key for "users" so that the users table refreshes.

Reply:
Done!